src/linear_algebra/tensor_product.lean

```diff
@@ -366,6 +365,14 @@ begin
   { intros t₁ t₂ ht₁ ht₂, exact submodule.add_mem _ ht₁ ht₂, },
 end
 
+lemma span_image2_eq_top {sM : set M} {sN : set N}
+  (hM : submodule.span R sM = ⊤) (hN : submodule.span R sN = ⊤) :
+  submodule.span R (set.image2 (tmul R) sM sN) = ⊤ :=
+begin
+  erw [← top_le_iff, ← submodule.map₂_span_span R (mk R M N), submodule.map₂_eq_span_image2,
+    ← span_tmul_eq_top, hM, hN], exact submodule.span_mono (λ _ ⟨m,n,h⟩, ⟨m,n,trivial,trivial,h⟩),
+end
```

Comment:
I think a nicer statement here would be that
```suggestion
lemma map₂_mk_top_top :
  submodule.map₂ (tensor_product.mk R M N) ⊤ ⊤ = (⊤ : submodule R (M ⊗[R] N)) =
```
which should follow trivially from `span_tmul_eq_top` above

Reply:
Hmm, yes I think I should prove in addition that for `M' : submodule M` and `N' : submodule N`, `M'.fg` and `N'.fg` imply that `(submodule.map₂ (tensor_product.mk R M N) M' N').fg`.

Reply:
Yes exactly. We already have that proof for mul, I expect it will generalize trivially to map2

Reply:
More generally we should prove `(submodule.map₂ f M' N').fg` for any bilinear map `f` when `M'` and N'` are finitely generated. Should be easy from `submodule.map₂_span_span`.